packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx

```diff
@@ -71,250 +59,73 @@ export const TreeNodeRenderer: React.ForwardRefExoticComponent<TreeNodeRendererP
-              onNodeKeyDown?.(node, !isSelected, event);
-            }
-          }}
-          onExpanded={(_, isExpanded) => {
-            expandNode(node.id, isExpanded);
-          }}
-          icon={getIcon ? getIcon(node) : undefined}
-          label={getLabel ? getLabel(node) : node.label}
-          sublabel={getSublabel ? getSublabel(node) : undefined}
-        >
-          <ButtonGroup className={cx("action-buttons", actionButtonsClassName)}>
-            {getHierarchyLevelDetails && node.isFiltered ? (
-              <IconButton
-                className="filtering-action-button"
-                styleType="borderless"
-                size="small"
-                label={localizedStrings.clearHierarchyLevelFilter}
-                onClick={(e) => {
-                  e.stopPropagation();
-                  getHierarchyLevelDetails(node.id)?.setInstanceFilter(undefined);
-                  applyFilterButtonRef.current?.focus();
-                }}
-              >
-                <SvgRemove />
-              </IconButton>
-            ) : null}
-            {onFilterClick && node.isFilterable && (filterButtonsVisibility !== "hide" || node.isFiltered) ? (
-              <IconButton
-                ref={applyFilterButtonRef}
-                className="filtering-action-button"
-                styleType="borderless"
-                size="small"
-                label={localizedStrings.filterHierarchyLevel}
-                onClick={(e) => {
-                  e.stopPropagation();
-                  const hierarchyLevelDetails = getHierarchyLevelDetails?.(node.id);
-                  hierarchyLevelDetails && onFilterClick(hierarchyLevelDetails);
-                }}
-              >
-                {node.isFiltered ? <SvgFilter /> : <SvgFilterHollow />}
-              </IconButton>
-            ) : null}
-          </ButtonGroup>
-        </TreeNode>
+        <TreeErrorRenderer node={node} getHierarchyLevelDetails={getHierarchyLevelDetails} reloadTree={reloadTree} onFilterClick={onFilterClick} ref={ref} />
       );
     }
 
-    if (node.type === "ResultSetTooLarge") {
-      return (
-        <ResultSetTooLargeNode
-          {...treeNodeProps}
-          ref={ref}
-          limit={node.resultSetSizeLimit}
-          onOverrideLimit={getHierarchyLevelDetails ? (limit) => getHierarchyLevelDetails(node.parentNodeId)?.setSizeLimit(limit) : undefined}
-          onFilterClick={
-            onFilterClick
-              ? () => {
-                  const hierarchyLevelDetails = getHierarchyLevelDetails?.(node.parentNodeId);
-                  hierarchyLevelDetails && onFilterClick(hierarchyLevelDetails);
-                }
-              : undefined
-          }
-        />
-      );
-    }
-
-    if (node.type === "NoFilterMatches") {
-      return <TreeNode {...treeNodeProps} ref={ref} label={localizedStrings.noFilteredChildren} isDisabled={true} onExpanded={/* c8 ignore next */ () => {}} />;
-    }
+    const isDisabled = false;
```

Comment:
Why is this always `false`?

Reply:
We need to figure out if we need a callback to determine state. As right now there is no way to control it from treeRenderer. For now removed isDisabled and is using `aria-disabled`.

Reply:
Looks like in old implementation it was also not possible to control `isDisabled` through `TreeRenderer`. It was possible only when using `TreeNodeRenderer` directly. I think same approach could be used in new implementation. 

Or maybe we do not need to have an ability to disable nodes. Only disable error nodes to make then non-interactable.